aspnetcore/fundamentals/openapi/aspnetcore-openapi.md

```diff
@@ -279,21 +279,46 @@ When no explicit annotation is provided, the framework attempts to determine the
 
 #### Describe response types
 
-<!-- TODO: Restructure this section to cover both controller-based and minimal API apps -->
+OpenAPI supports providing a description of the responses returned from an API. ASP.NET Core provides several strategies for setting the response metadata of an endpoint. Response metadata that can be set includes the status code, the type of the response body, and content type(s) of a response. Responses in OpenAPI may have additional metadata, such as description, headers, links, and examples. This additional metadata can be set with a [document transformer](#use-document-transformers) or [operation transformer](#use-operation-transformers).
+
+The specific mechanisms for setting response metadata depend on the type of app being developed.
+
+##### [Minimal APIs](#tab/minimal-apis)
+
+In Minimal API apps, ASP.NET Core can extract the response metadata from extension methods on the endpoint, attributes on the route handler, and the return type of the route handler.
```

Comment:
```suggestion
In Minimal API apps, ASP.NET Core can extract the response metadata added by extension methods on the endpoint, attributes on the route handler, and the return type of the route handler.
```

Something to make it more clear that these strategies add the metadata to the endpoint and are not the source of metadata themselves.